### ISSUE TITLE
cypress does not exit if reporter throws an exception

### ISSUE THREAD

**Issue by egrubbs** [cypress-io/cypress#5412]
### Current behavior:

`npx cypress run` will not exit if the reporter throws an exception in the `end` event handler.

It will display this stack trace, and then the process will be idle.
```
Error: fail fast
    at Runner.BadReporter.runner.on (.../bad-reporter.js:4:11)
    at emitOne (events.js:116:13)
    at Runner.emit (events.js:211:7)
    at Reporter.emit (/home/egrubbs/.cache/Cypress/3.4.1/Cypress/resources/app/packages/server/lib/reporter.js:237:55)
    at Object.server.startWebsockets.onMocha (/home/egrubbs/.cache/Cypress/3.4.1/Cypress/resources/app/packages/server/lib/project.js:300:22)
    at Socket.<anonymous> (/home/egrubbs/.cache/Cypress/3.4.1/Cypress/resources/app/packages/server/lib/socket.js:236:36)
    at emitTwo (events.js:126:13)
    at Socket.emit (events.js:214:7)
    at /home/egrubbs/.cache/Cypress/3.4.1/Cypress/resources/app/packages/socket/node_modules/socket.io/lib/socket.js:528:12
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
```

### Desired behavior:

Cypress should exit so continuous integration reports an error without waiting for a timeout.

### Steps to reproduce: (app code and test code)

Here is a reporter that will throw an exception in the `end` event handler.

**bad-reporter.js:**
```js
function BadReporter(runner, options) {
  runner.on('end', () => {
    throw new Error('fail fast');
  });
}

module.exports = BadReporter;
```
Run test with bad-reporter.js:
```
npx cypress run --reporter bad-reporter
```

### Versions

Ubuntu
Cypress:    3.4.1
Browser:    Electron 61 (headless)

**Comment by jennifer-shehane** [comment]
Hey @egrubbs, could you explain a bit behind what your initial problem and end goal was? That may helps us understand the scope and how applicable this issue is to our other users. This helps us prioritize work.

Are you writing a reporter yourself? Are you using a reporter that throws an error that makes Cypress hang? Thanks!

**Comment by egrubbs** [comment]
Hi @jennifer-shehane, I have written my own reporter to provide more info to Azure Pipelines. When Cypress doesn't exit, it can be confusing, because it appears as if it could be a network error that prevents the build server from publishing its test results.

**Comment by flotwig** [comment]
I thought this might be fixed by #5413, but the issue still appears to exist, even with that patch.
